crawl-specs.js

```diff
@@ -325,8 +343,19 @@ function crawlList(speclist, crawlOptions) {
  * @return {Promise<void>} The promise to have saved the data
  */
 function saveResults(crawlInfo, crawlOptions, data, path) {
-    return new Promise((resolve, reject) => {
-        fs.readFile(path, function(err, content) {
+  return Promise.all(
+    data.map(
+      spec =>
+        new Promise((resolve, reject) => {
+          if (spec.idl.idl) {
+            fs.writeFile(pathlib.dirname(path) + '/idl/' + getShortname(spec) + '.idl',
+                         spec.idl.idl,
+                         err => { if (err) return console.log(err); return resolve();});
+            delete spec.idl.idl;
+          } else resolve();
+        })))
+    .then(new Promise((resolve, reject) => {
```

Comment:
This does not seem to bug Node.js, but that is wrong. First parameter to `then` should be a `Function`, not a `Promise`, so this should rather be:
`.then(_ => new Promise((resolve, reject) => {`

(In practice, the code of that promise currently runs _before_ the previous `Promise.all` call and any data it could return when it resolves gets ignored, which has no consequence here, but not exactly what you'd expect from a chain of `then`....)
That still seems w